starknetdata/feeder/feeder.go

```diff
@@ -1,35 +1,28 @@
-package gateway
+package feeder
 
 import (
 	"context"
 	"errors"
 
-	"github.com/NethermindEth/juno/clients"
+	"github.com/NethermindEth/juno/clients/feeder"
 	"github.com/NethermindEth/juno/core"
 	"github.com/NethermindEth/juno/core/felt"
-	"github.com/NethermindEth/juno/utils"
 	"github.com/ethereum/go-ethereum/common"
 )
 
-type Gateway struct {
-	client *clients.GatewayClient
+type Feeder struct {
+	client *feeder.Client
 }
 
-func NewGateway(n utils.Network) *Gateway {
-	return &Gateway{
-		client: clients.NewGatewayClient(n.URL()),
-	}
-}
-
-func NewGatewayWithClient(client *clients.GatewayClient) *Gateway {
-	return &Gateway{
+func New(client *feeder.Client) *Feeder {
+	return &Feeder{
 		client: client,
 	}
 }
 
-// BlockByNumber gets the block for a given block number from the feeder gateway,
+// BlockByNumber gets the block for a given block number from the feeder,
 // then adapts it to the core.Block type.
-func (g *Gateway) BlockByNumber(ctx context.Context, blockNumber uint64) (*core.Block, error) {
+func (g *Feeder) BlockByNumber(ctx context.Context, blockNumber uint64) (*core.Block, error) {
```

Comment:
Since we've changed the name to `Feeder`, we should probably also change this

```suggestion
func (f *Feeder) BlockByNumber(ctx context.Context, blockNumber uint64) (*core.Block, error) {
```